vispy/visuals/text/text.py

```diff
@@ -559,6 +563,26 @@ def _prepare_draw(self, view):
             assert pos.shape[0] == self._vertices.size == len(_rot)
             self.shared_program['a_pos'] = pos
             self._pos_changed = False
+        if self._color_changed:
+            # now we promote color to the proper shape (varying)
+            text = self.text
+            if not isinstance(text, string_types):
+                repeats = [4 * len(t) for t in text]
+                text = ''.join(text)
+            else:
+                repeats = [4 * len(text)]
+            n_text = len(repeats)
+            color = self.color.rgba
+            if color.shape[0] < n_text:
+                color = np.repeat(color,
+                                  [1]*(len(color)-1) + [n_text-len(color)+1],
+                                  axis=0)
+            color = np.repeat(color[:n_text], repeats, axis=0)
```

Comment:
Would it be a bad idea to try to re-cache these new color arrays back in to `self._color`? Maybe not the one repeated for each RGB color but the one that is made the same length as the text object? Or maybe that would cause confusion for the user that `.color` is not the same as what they set it to? Thoughts?

Reply:
I personally have a bad experience with this when I used clim='auto' in the Image visual and when I tried to read it back I got an array. This caused problems in my PyQt application. It is not a big issue but I would keep the value as it was set. It is just my personal preference though.
From the other point of view - what would be the benefit of re-caching the new color arrays?

Reply:
I suppose I was concerned that some of these color calculations would be recalculated every time the text was changed and perhaps they didn't need to be. Just worried about performance.

Reply:
Right. It depends on the length of the text I guess. I could test this and we can decide based on the test results?

Reply:
If you have the time, that would be great. Reducing number of operations that are required pre-draw or on-draw is a benefit.

Reply:
I ran the test as follows - I created a long random text updated it fast with and without re-caching the color looking for drop in FPS but did not see any. So I don't think there is any significant performance problem if we recalculate color every time the text changes. Although I have to say the FPS drop pretty fast with larger text. But the bottleneck seems to be somewhere else.

Let me know if you think of another way to test this.

Reply:
That is good enough for me I think. It is worth merging this and getting more eyes on it. I'll try merging this with current master and see if that fixes the test failures.